db.go

```diff
@@ -250,6 +250,164 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 	return m
 }
 
+// DBReadOnly provides APIs for read only operations on a database.
+type DBReadOnly struct {
+	logger  log.Logger
+	dir     string
+	closers []io.Closer
+	mtx     sync.Mutex
+	closed  bool
+}
+
+// OpenDBReadOnly opens DB in the given directory for read only operations.
+func OpenDBReadOnly(dir string, l log.Logger) (*DBReadOnly, error) {
+	if _, err := os.Stat(dir); err != nil {
+		return nil, err
+	}
+
+	if l == nil {
+		l = log.NewNopLogger()
+	}
+
+	db := &DBReadOnly{
+		logger: l,
+		dir:    dir,
+	}
+
+	return db, nil
+}
+
+// Querier loads the wal and returns a new querier over the data partition for the given time range.
+// A goroutine must not handle more than one open Querier.
+func (db *DBReadOnly) Querier(mint, maxt int64) (Querier, error) {
+	blocksReaders, err := db.Blocks()
+	if err != nil {
+		return nil, err
+	}
+	blocks := make([]*Block, len(blocksReaders))
+	for i, b := range blocksReaders {
+		b, ok := b.(*Block)
```

Comment:
Why we do that?

Reply:
I mean why have interface if we literally have one implementation we require everywhere... ?

Reply:
And it's even us who implement `db.Blocks` so can we return `[]*Block` in `db.Blocks` then?

Reply:
hm , I am still unsure what you mean here.